runtime/compiler/control/CompilationThread.cpp

```diff
@@ -549,6 +549,10 @@ TR_YesNoMaybe TR::CompilationInfo::shouldActivateNewCompThread()
 #if defined(J9VM_OPT_JITSERVER)
       else if (getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT && JITServerHelpers::isServerAvailable())
          {
+         // If the available memory on the server is low, do not activate more threads
```

Comment:
I would move this piece of code up where the 'NO' decisions are taken. As it is now, we may take the path that leads to TR_yes above.